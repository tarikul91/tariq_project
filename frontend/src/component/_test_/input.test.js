import { render, screen, cleanup } from "@testing-library/react";
import Input from "component/input";
import renderer from "react-test-renderer"
afterEach(cleanup)
test('Input test',()=>{
    render(<Input className='form-control'  type='password' disabled/>)
    expect(screen.getByTestId('input-1')).toBeInTheDocument()
    expect(screen.getByTestId('input-1')).toHaveClass('form-control')
    expect(screen.getByTestId('input-1')).toBeDisabled()
    expect(screen.getByTestId('input-1')).toHaveAttribute('type', 'password')
})
it('input snapshot',()=>{
    const tree = renderer.create(<Input defaultValue='email' className='form-control'/>).toJSON()
    expect(tree).toMatchSnapshot()
})