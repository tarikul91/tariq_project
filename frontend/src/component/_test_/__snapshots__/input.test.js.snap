// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input snapshot 1`] = `
<input
  className="form-control"
  data-testid="input-1"
  defaultValue="email"
/>
`;
